lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -391,6 +391,13 @@ func (c *ContainerProxy) AddVolumesToContainer(handle string, config types.Conta
 	}
 	log.Infof("Finalized volume list: %#v", volList)
 
+	// remove overlapping specified volume entries from the config.Config.Volumes map since they are superseding the anonymous behavior of image volumes.
+	for _, v := range config.HostConfig.Binds {
+		fields := strings.SplitN(v, ":", 2)
+		destination := fields[1]
```

Comment:
Should we do a len check for `fields` here for defensive purposes?

Reply:
Yes we can, in general this field should honestly always have this split input wise. But defensively we should definitely log if the lengths are wrong and return an error potentially. The `HostConfig.Binds` indicates that a call such as `docker -v my-volume:/goes/right/here` another way for docker to populate this field is `docker -v /this/is/my/data:/it/goes/here` in both cases to end up in the binds metadata you should absolutely meet the criteria for this split. I think it would likely indicate a regression upstream from here if the length check failed. Some thought into the logging message should be considered. 

From a docker perspective, a `docker create` option should rarely fail and we should work our hardest to not fail out. in this case though, I would find it hard for us to continue since it would be a case of incorrect metadata passed to us. I am not really sure how we would excercise this check outside of a unit test potentially. I will look at a way to ensure this does not regress, but I am not aware of a way to force that situation from docker client perspective.